core_impl/src/main/java/com/google/instrumentation/stats/MeasurementDescriptorToViewMap.java

```diff
@@ -38,29 +41,46 @@
   private final Multimap<MeasurementDescriptor.Name, MutableView> mutableMap =
       HashMultimap.<MeasurementDescriptor.Name, MutableView>create();
 
-  /** Returns a {@link View} corresponding to the given {@link ViewDescriptor}. */
-  synchronized View getView(ViewDescriptor viewDescriptor, Clock clock) {
-    MutableView view = getMutableView(viewDescriptor);
+  @GuardedBy("this")
+  private final Map<ViewDescriptor.Name, ViewDescriptor> registeredViews =
+      new HashMap<ViewDescriptor.Name, ViewDescriptor>();
+
+  /** Returns a {@link View} corresponding to the given {@link ViewDescriptor.Name}. */
+  synchronized View getView(ViewDescriptor.Name viewName, Clock clock) {
+    MutableView view = getMutableView(viewName);
     return view == null ? null : view.toView(clock);
   }
 
-  private synchronized MutableView getMutableView(ViewDescriptor viewDescriptor) {
+  @Nullable
+  private synchronized MutableView getMutableView(ViewDescriptor.Name viewName) {
+    ViewDescriptor viewDescriptor = registeredViews.get(viewName);
+    if (viewDescriptor == null) {
+      return null;
+    }
     Collection<MutableView> views =
         mutableMap.get(viewDescriptor.getMeasurementDescriptor().getMeasurementDescriptorName());
     for (MutableView view : views) {
-      if (view.getViewDescriptor().equals(viewDescriptor)) {
+      if (view.getViewDescriptor().getViewDescriptorName().equals(viewName)) {
         return view;
       }
     }
-    return null;
+    throw new AssertionError("registeredViews and mutableMap contain different views: "
```

Comment:
Will this ever be reached? (since we already performed check in `registerView()`) 
Also since users don't know implementation details on `registeredViews` and `mutableMap`, this error message could be changed to something like `"View mismatch: Not recording stats for view: " + viewName`